### PR TITLE
Homebrew formula generator

### PR DESCRIPTION
**Pull request by alexsaveliev**
Added script and template used to generate Homebrew formula

### CODE REVIEW

File: brew/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+![se7en](http://www.flutecrate.com/uploads/1/0/2/0/10200817/5243300_orig.jpg)
```
Comment (slimsag) [inline brew/README.md]: 😆

**Comment by slimsag** [comment]
This looks pretty cool! Is the goal to get go-langserver into homebrew for VS Code users ? or for some other use-case (just curious / out of the loop! 😄 )

I only looked at the README, since I don't know Bash super well. I'll leave final choice up to @keegancsmith but looks okay to me!

File: brew/go-langserver.rb.template [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  test do
+    # Set up fake GOROOT and create test Go project
+    mkdir_p testpath/"gopath/src/test"
+    mkdir_p testpath/"goroot"
+    ENV["GOPATH"] = "#{testpath}/gopath"
+    ENV["GOROOT"] = "#{testpath}/goroot"
+    (testpath/"gopath/src/test/p/a.go").write("package p; func A() {}")
+    # Invoke initialize, hover, and finally exit requests and make sure that LSP server returns proper data
+    init_req = "{\"id\":0,\"method\":\"initialize\",\"params\":{\"rootPath\":\"file://#{testpath}/gopath/src/test/p\"}}"
```
Comment (keegancsmith) [inline brew/go-langserver.rb.template]: if you use single quoted strings, you won't need to escape `"`.

**Comment by alexsaveliev** [comment]
@slimsag 
> Is the goal to get go-langserver into homebrew for VS Code users ? 

Yes, there is a [Trello card](https://trello.com/c/1jjYLVBC/1-i-can-install-the-go-lsp-on-my-mac-using-homebrew-1) :)

File: brew/go-langserver.rb.template (hunk 2 above)
Comment (alexsaveliev) [inline brew/go-langserver.rb.template]: > if you use single quoted strings, you won't need to escape ".

I did it to enable string interpolation `#{testpath}`

**Comment by slimsag** [comment]
@alexsaveliev @keegancsmith status?

**Comment by alexsaveliev** [comment]
@slimsag hey, I don't have permissions to commit/merge into this repository. Could you please merge my changes?

**Comment by keegancsmith** [comment]
@alexsaveliev sorry! didn't realise that.

**Pull request merged**